src/coreclr/jit/optimizer.cpp

```diff
@@ -4493,6 +4493,15 @@ PhaseStatus Compiler::optInvertLoops()
         fgModified = false;
     }
 
+    // optInvertWhileLoop can cause IR changes even if it does not modify
+    // the flow graph. It calls gtPrepareCost which can cause operand swapping.
+    // Work around this for now.
+    //
+    // Note phase status only impacts dumping and checking done post-phase,
+    // it has no impact on a release build.
+    //
+    madeChanges = true;
+
```

Comment:
You could also add an additional state MODIFIED_TREE here instead of using MODIFIED_EVERYTHING

Reply:
Right -- I originally had finer-grained status in mind, but wasn't quite sure what the right breakout should be, so just have everything/nothing for now.